tests/kernel/threads/usage/src/task.c

```diff
@@ -0,0 +1,384 @@
+
+K_SEM_DEFINE(sem_thread2, 0, 1);
+K_SEM_DEFINE(sem_thread1, 0, 1);
+
+K_THREAD_STACK_DEFINE(k_stack_thread1, STACKSIZE);
+K_THREAD_STACK_DEFINE(k_stack_thread2, STACKSIZE);
+
+__kernel struct k_thread k_thread_thread2;
+__kernel struct k_thread k_thread_thread1;
+k_tid_t thread2_id;
+k_tid_t thread1_id;
+
+/**
+ *
+ * @brief Test thread_id_get() and k_thread_priority_get() macros
+ *
+ */
+
+void thread_macros_test(k_tid_t thread_id, int thread_prio)
+{
+	k_tid_t thread_id_value;
+	int thread_prio_value;
+
+	thread_id_value = k_current_get();
+	zassert_equal(thread_id_value, thread_id,
+		      "k_current_get() returned %d, not %d\n",
+		      (u32_t)thread_id_value, (u32_t)thread_id);
+
+	thread_prio_value = k_thread_priority_get(k_current_get());
+	zassert_equal(thread_prio_value, thread_prio,
+		      "k_thread_priority_get() returned %d, not %d\n",
+		      (u32_t)thread_prio_value, (u32_t)thread_prio);
+}
+
+/**
+ *
+ * @brief thread2 portion to test setting the priority
+ *
+ */
+
+void thread2_set_prio_test(void)
+{
+	/* lower thread2 priority by 5 */
+	k_sem_take(&sem_thread2, K_FOREVER);
+	thread2_data = k_thread_priority_get(k_current_get());
+	k_sem_give(&sem_thread1);
+
+	/* raise thread2 priority by 10 */
+	k_sem_take(&sem_thread2, K_FOREVER);
+	thread2_data = k_thread_priority_get(k_current_get());
+	k_sem_give(&sem_thread1);
+
+	/* restore thread2 priority */
+	k_sem_take(&sem_thread2, K_FOREVER);
+	thread2_data = k_thread_priority_get(k_current_get());
+	k_sem_give(&sem_thread1);
+}
+
+/**
+ *
+ * @brief Test the k_thread_priority_set() API
+ *
+ */
+
+void thread_set_prio_test(void)
+{
+	int  rv;
+
+	/* Lower the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO + 2);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO + 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO + 2, rv);
+
+	/* Raise the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO - 2);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO - 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO - 2, rv);
+
+	/* Restore the priority of the current thread (thread1) */
+	k_thread_priority_set(thread1_id, THREAD1_PRIO);
+	rv = k_thread_priority_get(k_current_get());
+	zassert_equal(rv, THREAD1_PRIO,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD1_PRIO, rv);
+
+	/* Lower the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO + 2);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
+	zassert_equal(thread2_data, THREAD2_PRIO + 2,
+		      "Expected priority to be changed to %d, not %d\n",
+		      THREAD2_PRIO + 2, thread2_data);
+
+	/* Raise the priority of thread2 */
+	k_thread_priority_set(thread2_id, THREAD2_PRIO - 2);
+	k_sem_give(&sem_thread2);
+	k_sem_take(&sem_thread1, K_FOREVER);
```

Comment:
do we really need two semaphores here? i think one would be sufficient to synchronize b/w two threads, please recheck once.